lib/bindings/langs/swift/Sources/BreezSDKLiquid/Task/SwapUpdated.swift

```diff
@@ -31,6 +34,52 @@ class SwapUpdatedTask : TaskProtocol {
             self.onShutdown()
             throw e
         }
+
+        startPolling(liquidSDK: liquidSDK)
+    }
+
+    func startPolling(liquidSDK: BindingLiquidSdk) {
+        pollingTimer = Timer.scheduledTimer(withTimeInterval: pollingInterval, repeats: true) { [weak self] _ in
+            guard let self = self else { return }
+            do {
+                guard let request = self.request else {
+                    self.stopPolling(withError: NSError(domain: "SwapUpdatedTask", code: -1, userInfo: [NSLocalizedDescriptionKey: "Missing swap updated request"]))
+                    return
+                }
+
+                if let payment = try liquidSDK.getPayment(req: .swapId(swapId: request.id)) {
+                    switch payment.status {
+                    case .complete:
+                        onEvent(e: SdkEvent.paymentSucceeded(details: payment))
+                        self.stopPolling()
+                    case .waitingFeeAcceptance:
+                        onEvent(e: SdkEvent.paymentWaitingFeeAcceptance(details: payment))
+                        self.stopPolling()
+                    case .pending:
+                        if paymentClaimIsBroadcasted(details: payment.details) {
+                            onEvent(e: SdkEvent.paymentWaitingConfirmation(details: payment))
+                            self.stopPolling()
+                        }
+                    default:
+                        break
+                    }
+                }
+            } catch {
+                self.stopPolling(withError: error)
+            }
+        }
+
+        pollingTimer?.fire()
+    }
+    
+    private func stopPolling(withError error: Error? = nil) {
```

Comment:
Do we also need to stop polling if onShutdown is called?

Reply:
Same as above